lib/ansible/plugins/connection/psrp.py

```diff
@@ -553,26 +553,24 @@ def _parse_pipeline_result(self, pipeline):
         # TODO: figure out a better way of merging this with the host output
         stdout_list = []
         for output in pipeline.output:
-            # not all pipeline outputs can be casted to a string, we will
-            # create our own output based on the properties if that is the
-            # case+
-            try:
-                output_msg = str(output)
-            except TypeError:
-                if isinstance(output, GenericComplexObject):
-                    obj_lines = output.property_sets
-                    for key, value in output.adapted_properties.items():
-                        obj_lines.append("%s: %s" % (key, value))
-                    for key, value in output.extended_properties.items():
-                        obj_lines.append("%s: %s" % (key, value))
-                    output_msg = "\n".join(obj_lines)
-                else:
-                    output_msg = ""
+            # Not all pipeline outputs are a string or contain a __str__ value,
+            # we will create our own output based on the properties of the
+            # complex object if that is the case.
+            if isinstance(output, GenericComplexObject) and output.to_string is None:
```

Comment:
(Also, is this the way to check for to_string?  I don't know the API so I don't know if it really creates the attribute but sets it to None or if it omits the attribute altogether.)

Reply:
`to_string` always exists, it's just set to `None` by default https://github.com/jborean93/pypsrp/blob/master/pypsrp/complex_objects.py#L109. It will get populated if the `<ToString></ToString>` element is populated on the XML being deserialized from the server. In the first case, `to_string` is `None` so we need to manually build the string representation from the properties otherwise we just call `str()` which works for a `GenericComplexObject` as well as any other non generic `ComplexObjects`.